episodes/03-explore-data.Rmd

```diff
@@ -0,0 +1,278 @@
+
+
+# [Data frame Manipulation with dplyr](https://datacarpentry.org/r-intro-geospatial/06-dplyr/index.html) 
+
+## Select
+Let's start manipulating the data. 
+
+First, we will adapt our data set, by keeping only the columns we're interested in, using the `select()` function from the `dplyr` package:
+
+```{r dplyr-select}
+year_country_gdp <- select(gapminder, year, country, gdpPercap) 
+
+head(year_country_gdp)
+
+```
+
+## Pipe
+Now, this is not the most common notation when working with `dplyr` package. `dplyr` offers an operator `%>%` called a pipe, which allows you build up very complicated commands in a readable way.
+
+
+In newer installation of `R` you can also find a notation `|>` . This pipe works in a similar way. The main difference is that you don't need to load any packages to have it available.
+
+
+The `select()` statement with pipe would look like that:
+
+```{r dplyr-pipe}
+
+year_country_gdp <- gapminder %>% 
+  select(year,country,gdpPercap)
+
+head(year_country_gdp)
+
+```
+
+First we define data set, then - with the use of pipe we pass it on to the `select()` function. This way we can chain multiple functions together, which we will be doing now. 
+
+## Filter
+
+We already know how to select only the needed columns. But now, we also want to filter the data set via certain conditions with `filter()` function. Instead of doing it in separate steps, we can do it all together. 
+
+In the `gapminder` data set, we want to see the results from outside of Europe for the 21st century. 
+```{r}
+year_country_gdp_euro <- gapminder %>% 
+  filter(continent != "Europe" & year >= 2000) %>% # & operator (AND) - both conditions must be met
+  select(year, country, gdpPercap)
+
+head(year_country_gdp_euro)
+```
+Let's now find all the observations from Eurasia: 
+
+```{r}
+year_country_gdp_eurasia <- gapminder %>% 
+  filter(continent == "Europe" | continent == "Asia") %>% # I operator (OR) - one of the conditions must be met
+  select(year, country, gdpPercap)
+
+head(year_country_gdp_eurasia)
+```
+
+###  Exercise 1
+
+<div class="alert alert-info">
+<strong>Challenge</strong>
+Write a single command (which can span multiple lines and includes pipes) that will produce a data frame that has the values for life expectancy, country and year, only for Eurasia. How many rows does your data frame have and why? 
+
+
+<strong>Solution</strong>
+
+</div>
+
+
+```{r ex5, class.source="bg-info"}
+year_country_lifeExp_Africa <- gapminder %>%
+                           filter(continent=="Africa"  ) %>%
+                           select(year,country,lifeExp)
+
+
+nrow(year_country_lifeExp_Africa)
+``` 
+
+
+
+## Group and summarize
+So far, we have created a data frame for one of the continents represented in the `gapminder` data set. But often instead of doing that, we would like to know statistics about all of the continents, presented by group.
```

Comment:
```suggestion
So far, we have provided summary statistics on the whole dataset or a particular subset. But often instead of doing that, we would like to know statistics about all continents, presented by group.
```

Reply:
I had to change the introduction line as it did not fit anymore with the previous exercise